File: lib/domain/internal_jobs/ports/scheduler_weather_fetch_schedule_port.rb
# frozen_string_literal: true

module Domain
  module InternalJobs
    module Ports
      module SchedulerWeatherFetchSchedulePort
        def schedule_fetch(farm_id:, latitude:, longitude:, start_date:, end_date:, delay_secs:)
          raise NotImplementedError, "#{self.class}#schedule_fetch"
        end

        def flush
          raise NotImplementedError, "#{self.class}#flush"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/internal_jobs/ports/scheduler_weather_fetch_schedule_port.rs
use time::Date;

/// Schedules in-process `FetchWeatherDataJob` equivalents with staggered delays.
pub trait SchedulerWeatherFetchSchedulePort: Send + Sync {
    fn schedule_fetch(
        &self,
        farm_id: i64,
        latitude: f64,
        longitude: f64,
        start_date: Date,
        end_date: Date,
        delay_secs: u64,
    );

    /// Flush queued steps to the job dispatcher (no-op if nothing scheduled).
    fn flush(&self);
}
